src/System.IO.FileSystem/src/System/IO/UnixFileStream.cs

```diff
@@ -957,7 +957,7 @@ public override long Seek(long offset, SeekOrigin origin)
         /// <returns>The new position in the stream.</returns>
         private long SeekCore(long offset, SeekOrigin origin)
         {
-            Contract.Assert(!_fileHandle.IsClosed && _canSeek);
+            Contract.Assert(!_fileHandle.IsClosed && CanSeek);
```

Comment:
not _parent.  SeekCore is called during construction and *should not* call any virtual members.